packages/atomic/src/components/commerce/atomic-commerce-interface/atomic-commerce-interface.tsx

```diff
@@ -0,0 +1,480 @@
+      return;
+    }
+
+    this.commonInterfaceHelper.onAnalyticsChange();
+  }
+
+  @Watch('language')
+  public updateLanguage() {
+    if (!this.commonInterfaceHelper.engineIsCreated(this.engine)) {
+      return;
+    }
+    this.commonInterfaceHelper.onLanguageChange();
+  }
+
+  @Watch('iconAssetsPath')
+  public updateIconAssetsPath() {
+    this.store.set('iconAssetsPath', this.iconAssetsPath);
+  }
+
+  public disconnectedCallback() {
+    this.unsubscribeUrlManager();
+    this.unsubscribeSearchStatus();
+    window.removeEventListener('hashchange', this.onHashChange);
+  }
+
+  @Listen('atomic/initializeComponent')
+  public handleInitialization(event: InitializeEvent) {
+    this.commonInterfaceHelper.onComponentInitializing(event);
+  }
+
+  @Listen('atomic/scrollToTop')
+  public scrollToTop() {
+    const scrollContainerElement = document.querySelector(this.scrollContainer);
+    if (!scrollContainerElement) {
+      this.bindings.engine.logger.warn(
+        `Could not find the scroll container with the selector "${this.scrollContainer}". This will prevent UX interactions that require a scroll from working correctly. Please check the CSS selector in the scrollContainer option`
+      );
+      return;
+    }
+
+    scrollContainerElement.scrollIntoView({behavior: 'smooth'});
+  }
+
+  /**
+   * Initializes the connection with the headless search engine using options for accessToken (required), organizationId (required), renewAccessToken, organizationEndpoints (recommended), and platformUrl (deprecated).
+   */
+  @Method() public initialize(options: CommerceInitializationOptions) {
+    return this.internalInitialization(() => this.initEngine(options));
+  }
+
+  @Method() public initializeWithEngine(engine: CommerceEngine) {
+    return this.internalInitialization(() => {
+      this.engine = engine;
+    });
+  }
+
+  /**
+   * Initializes the connection with an already preconfigured [headless search engine](https://docs.coveo.com/en/headless/latest/reference/search/), as opposed to the `initialize` method, which will internally create a new search engine instance.
+   * This bypasses the properties set on the component, such as analytics, searchHub, pipeline, language, timezone & logLevel.
+   */
+  @Method() public initializeWithCommerceEngine(engine: CommerceEngine) {
+    return this.internalInitialization(() => (this.engine = engine));
+  }
+
+  /**
+   *
+   * Executes the first search and logs the interface load event to analytics, after initializing connection to the headless search engine.
+   */
+  @Method() public async executeFirstSearch() {
+    if (!this.commonInterfaceHelper.engineIsCreated(this.engine)) {
+      return;
+    }
+
+    if (!this.initialized) {
+      console.error(
+        'You have to wait until the "initialize" promise is fulfilled before executing a search.',
+        this.host
+      );
+      return;
+    }
+
+    if (this.localizationCompatibilityVersion !== 'v4') {
```

Comment:
Related to above comment: Latest mode should be assumed, no warning will be necessary.